diff -u --recursive kodi-14.0rc1-vanilla/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp kodi-14.0rc1/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp
--- kodi-14.0rc1-vanilla/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2014-12-02 19:21:14.165909454 -0500
+++ kodi-14.0rc1/xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxFFmpeg.cpp	2014-12-02 19:27:21.705921942 -0500
@@ -518,10 +518,6 @@
 
 void CDVDDemuxFFmpeg::Flush()
 {
-  // naughty usage of an internal ffmpeg function
-  if (m_pFormatContext)
-    av_read_frame_flush(m_pFormatContext);
-
   m_currentPts = DVD_NOPTS_VALUE;
 
   m_pkt.result = -1;
Only in kodi-14.0rc1/xbmc/cores/dvdplayer/DVDDemuxers: DVDDemuxFFmpeg.cpp.orig
Only in kodi-14.0rc1/xbmc/cores/dvdplayer/DVDDemuxers: DVDDemuxFFmpeg.cpp.rej
