--- configure.in.orig	2010-08-20 00:21:05.638518410 -0400
+++ configure.in	2010-08-20 00:20:04.949267903 -0400
@@ -1866,22 +1866,6 @@
   fi
 ], [0])
 
-XB_CONFIG_MODULE([lib/libmodplug], [
-  if test "$host_vendor" = "apple" ; then
-    CFLAGS="$CFLAGS -fno-common" \
-    CXXFLAGS="$CXXFLAGS -fno-common" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic --disable-shared --enable-static
-  fi
-], [$HOST_NOT_APPLE])
-
-XB_CONFIG_MODULE([lib/libmicrohttpd], [
-  if test "$host_vendor" = "apple" ; then
-    CFLAGS="$CFLAGS -fno-common" \
-    CXXFLAGS="$CXXFLAGS -fno-common" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic --disable-shared --enable-static --with-libgcrypt-prefix=/dev/null
-  fi
-], [$HOST_NOT_APPLE])
-
 XB_CONFIG_MODULE([lib/cpluff], [
   if test "$host_vendor" = "apple" ; then
     ./configure --disable-nls --with-libiconv-prefix=/opt/local --with-libintl-prefix=/opt/local --without-readline CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS"
