--- xbmc-9.11.orig/tools/Linux/xbmc.sh.in	2009-12-06 11:44:24.000000000 +0100
+++ xbmc-9.11/tools/Linux/xbmc.sh.in	2010-01-20 17:26:39.000000000 +0100
@@ -5,7 +5,7 @@
   find $1 -maxdepth $2 -name 'core*' -exec \
       echo "=====>  Core file: {}" 2> /dev/null >> $FILE \; -exec \
       echo "        ========================================= " 2> /dev/null >> $FILE \; -exec \
-      gdb /usr/share/xbmc/xbmc.bin --core={} --batch -ex "thread apply all bt" 2> /dev/null >> $FILE \; -exec \
+      /usr/bin/gdb @libdir@/xbmc/xbmc.bin --core={} --batch -ex "thread apply all bt" 2> /dev/null >> $FILE \; -exec \
       rm -f {} \;
 }
 
@@ -37,7 +37,7 @@
   if [ $XBMC_HOME ]; then
     BASEDIR=$XBMC_HOME
   else
-    BASEDIR=@prefix@/share/xbmc/
+    BASEDIR=@libdir@/xbmc/
   fi
   single_stacktrace $BASEDIR 5
   # find in user xbmc dir
@@ -63,11 +63,13 @@
 }
 
 # Set XBMC_HOME if xbmc.bin is a symlink
-if [ -L @prefix@/share/xbmc/xbmc.bin ]; then
-  export XBMC_HOME="@prefix@/share/xbmc"
+if [ -L @libdir@/xbmc/xbmc.bin ]; then
+  export XBMC_HOME="@libdir@/xbmc"
 fi
 
-python @prefix@/share/xbmc/FEH.py "$@"
+# export XBMC_HOME=@libdir@/xbmc
+ 
+python @libdir@/xbmc/FEH.py "$@"
 if [ $? -ne 0 ]; then
   exit
 fi
@@ -76,7 +78,7 @@
 while (( $LOOP ))
 do
   LOOP=0
-  @prefix@/share/xbmc/xbmc.bin "$@"
+  @libdir@/xbmc/xbmc.bin "$@"
   RET=$?
   if (( $RET == 65 ))
   then
