--- configure.in.orig	2010-08-20 00:17:55.000000000 -0400
+++ configure.in	2010-08-20 00:19:27.621394427 -0400
@@ -1736,14 +1736,6 @@
   fi
 ], [$HOST_NOT_APPLE])
 
-XB_CONFIG_MODULE([xbmc/lib/cximage-6.0/zlib], [
-  if test "$host_vendor" = "apple" ; then
-    export MACOSX_DEPLOYMENT_TARGET=10.4
-    CFLAGS="$CFLAGS -fno-common" \
-    ./configure
-  fi
-], [0])
-
 XB_CONFIG_MODULE([lib/libass], [
   if test "$host_vendor" = "apple" ; then
     CFLAGS="$CFLAGS -fno-common" \
