--- configure.in.orig	2010-03-06 04:48:49.233198314 -0500
+++ configure.in	2010-03-06 04:51:02.297020657 -0500
@@ -990,7 +990,6 @@
              xbmc/lib/libexif/Makefile \
              xbmc/lib/libXDAAP/Makefile \
              xbmc/lib/libhdhomerun/Makefile \
-             xbmc/lib/libGoAhead/Makefile \
              xbmc/lib/libshout/Makefile \
              xbmc/lib/libsquish/Makefile \
              xbmc/lib/libid3tag/Makefile \
--- Makefile.in.orig	2010-03-06 04:49:38.948896140 -0500
+++ Makefile.in	2010-03-06 04:50:24.628895410 -0500
@@ -29,7 +29,6 @@
 	xbmc/karaoke \
 	xbmc/lib/libcmyth \
 	xbmc/lib/libhts \
-	xbmc/lib/libGoAhead \
 	xbmc/lib/libPython \
 	xbmc/lib/libPython/xbmcmodule \
 	xbmc/lib/libRTMP \
@@ -219,8 +218,6 @@
 	$(MAKE) -C xbmc/lib/libXBMS
 xbmc/lib/libXDAAP/libxdaap-@ARCH@.a: force
 	$(MAKE) -C xbmc/lib/libXDAAP
-xbmc/lib/libGoAhead/libGoAhead-@ARCH@.a: force
-	$(MAKE) -C xbmc/lib/libGoAhead
 xbmc/lib/sqLite/sqllite.a: force
 	$(MAKE) -C xbmc/lib/sqLite
 xbmc/lib/UnrarXLib/UnrarXLib.a: force
@@ -315,8 +312,7 @@
 	xbmc/lib/libXBMS/libxbms-@ARCH@.a \
 	xbmc/lib/libRTV/librtv-@ARCH@.a \
 	xbmc/lib/libXDAAP/libxdaap-@ARCH@.a \
-	xbmc/lib/libshout/libshout-@ARCH@.a \
-	xbmc/lib/libGoAhead/libGoAhead-@ARCH@.a
+	xbmc/lib/libshout/libshout-@ARCH@.a
 
 # platform independend objects
 OBJSXBMC=	\
@@ -355,7 +351,6 @@
 	xbmc/lib/libUPnP/libupnp-@ARCH@.a \
 	xbmc/lib/libXBMS/libxbms-@ARCH@.a \
 	xbmc/lib/libXDAAP/libxdaap-@ARCH@.a \
-	xbmc/lib/libGoAhead/libGoAhead-@ARCH@.a \
 	xbmc/lib/sqLite/sqllite.a \
 	xbmc/lib/libsquish/libsquish-@ARCH@.a \
 	xbmc/screensavers/screensaver.a \
--- guilib/system.h.orig	2010-03-06 05:42:46.227020727 -0500
+++ guilib/system.h	2010-03-06 05:44:35.429895860 -0500
@@ -44,7 +44,7 @@
 #define HAS_UPNP
 #define HAS_VIDEO_PLAYBACK
 #define HAS_VISUALISATION
-#define HAS_WEB_SERVER
+/* undef HAS_WEB_SERVER */
 
 #define HAS_AC3_CODEC
 #define HAS_DTS_CODEC
