diff -Naur xbmc-9.11-8.2/xbmc/utils/Weather.cpp xbmc-9.11-8.3/xbmc/utils/Weather.cpp
--- xbmc-9.11-8.2/xbmc/utils/Weather.cpp	2009-11-12 08:41:55.000000000 +0100
+++ xbmc-9.11-8.3/xbmc/utils/Weather.cpp	2010-01-22 10:56:40.000000000 +0100
@@ -399,7 +399,7 @@
 
         GetString(pOneDayElement, "low", iTmpStr, "");
         if (iTmpStr == "N/A")
-          m_info.forecast[i].m_high = "";
+          m_info.forecast[i].m_low = "";
         else
         {
           CTemperature temp=CTemperature::CreateFromCelsius(atoi(iTmpStr));
@@ -417,8 +417,12 @@
             m_info.forecast[i].m_icon.Format("%s128x128/na.png", WEATHER_BASE_PATH);
           else
             m_info.forecast[i].m_icon.Format("%s128x128/%s.png", WEATHER_BASE_PATH, iTmpStr);
-
-          GetString(pDayTimeElement, "t", m_info.forecast[i].m_overview, "");
+            
+          GetString(pDayTimeElement, "t", iTmpStr, ""); //string cause i've seen it return N/A
+          if (iTmpStr == "N/A")
+            m_info.forecast[i].m_overview = "";
+          else
+            m_info.forecast[i].m_overview = iTmpStr;
           LocalizeOverview(m_info.forecast[i].m_overview);
         }
 
