diff -Naur xbmc-9.11-11.0/xbmc/lib/libid3tag/libid3tag/metadata.c xbmc-9.11-11.1/xbmc/lib/libid3tag/libid3tag/metadata.c
--- xbmc-9.11-11.0/xbmc/lib/libid3tag/libid3tag/metadata.c	2009-10-13 02:48:48.000000000 +0200
+++ xbmc-9.11-11.1/xbmc/lib/libid3tag/libid3tag/metadata.c	2010-02-03 07:33:21.000000000 +0100
@@ -23,6 +23,9 @@
 #  include "config.h"
 # endif
 
+#include <string.h>
+#include <stdlib.h>
+
 # include "metadata.h"
 # include "global.h"
 # include "ucs4.h"
