diff -Naur xbmc-9.11-10.0/tools/TexturePacker/Makefile.in xbmc-9.11-10.1/tools/TexturePacker/Makefile.in
--- xbmc-9.11-10.0/tools/TexturePacker/Makefile.in	2009-10-04 05:55:21.000000000 +0200
+++ xbmc-9.11-10.1/tools/TexturePacker/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -21,4 +21,4 @@
 include ../../Makefile.include
 
 $(TARGET): $(OBJS)
-	$(CXX) $(OBJS) $(LDFLAGS) $(LIBS) -o $(TARGET)
+	$(CXX) $(CXXFLAGS) $(OBJS) $(LDFLAGS) $(LIBS) -o $(TARGET)
diff -Naur xbmc-9.11-10.0/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in xbmc-9.11-10.1/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in	2010-01-25 14:29:23.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -48,12 +48,12 @@
 else
 
 $(SYSDIR)/libdvdcss-$(ARCH).so: $(WRAPPER) libdvdcss/src/.libs/libdvdcss.a
-	$(CC) -o $@ $(LDFLAGS) --soname,$@ \
+	$(CC) $(CFLAGS) -o $@ $(LDFLAGS) --soname,$@ \
 		libdvdcss/src/*.o \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
 $(SYSDIR)/libdvdnav-$(ARCH).so: $(WRAPPER) $(DVDCSS_A) libdvdnav/obj/libdvdnav.a
-	$(CC) -o $@ $(LDFLAGS) --soname,$@ $(DVDCSS_O) libdvdnav/obj/*.o \
+	$(CC) $(CFLAGS) -o $@ $(LDFLAGS) --soname,$@ $(DVDCSS_O) libdvdnav/obj/*.o \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
 endif
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/AC3Codec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/AC3Codec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/AC3Codec/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/AC3Codec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -1,5 +1,5 @@
 ARCH=@ARCH@
-CFLAGS=-D_LINUX -fPIC -Iinclude -Ivc++
+CFLAGS=-D_LINUX -fPIC -Iinclude
 ifeq ($(findstring osx,$(ARCH)), osx)
     export MACOSX_DEPLOYMENT_TARGET=10.4
     CFLAGS+=-fno-common -isysroot /Developer/SDKs/MacOSX10.4u.sdk -mmacosx-version-min=10.4
@@ -20,7 +20,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) \
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) \
 		`cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/ADPCMCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/ADPCMCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/ADPCMCodec/Makefile.in	2009-03-22 19:56:49.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/ADPCMCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -1,7 +1,7 @@
 ARCH=@ARCH@
 OBJS=uXboxAdpcmDecoder.o ADPCMDll.o
-CFLAGS +=-D_LINUX -fPIC -O2
-CXXFLAGS += -D_LINUX -fPIC -O2
+CFLAGS +=-D_LINUX -fPIC
+CXXFLAGS += -D_LINUX -fPIC
 
 SLIB=../../../../system/players/paplayer/adpcm-@ARCH@.so
 
@@ -11,7 +11,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -shared -o $@ *.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
+	$(CC) $(CFLAGS) -shared -o $@ *.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
 include ../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/FLACCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/FLACCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/FLACCodec/Makefile.in	2009-10-07 10:28:05.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/FLACCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -19,12 +19,12 @@
 else
 ifneq (@USE_EXTERNAL_LIBFLAC@,1)
 $(FLAC_LIB): flac-1.2.1/src/libFLAC/.libs/libFLAC.a
-	$(CC) -shared -fpic -o $@ -rdynamic \
+	$(CC) $(CFLAGS) -shared -fpic -o $@ -rdynamic \
 	-Wl,--whole-archive flac-1.2.1/src/libFLAC/.libs/libFLAC.a -Wl,--no-whole-archive \
 	`cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 else
 $(FLAC_LIB):
-	$(CC) -shared -fpic -o $@ -rdynamic -lFLAC \
+	$(CC) $(CFLAGS) -shared -fpic -o $@ -rdynamic -lFLAC \
 	`cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 endif
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/GYMCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/GYMCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/GYMCodec/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/GYMCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -15,7 +15,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CXX) -shared -o $@ *.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
+	$(CXX) $(CXXFLAGS) -shared -o $@ *.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 
 endif
 
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/MACDll/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/MACDll/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/MACDll/Makefile.in	2009-10-07 10:28:05.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/MACDll/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -65,7 +65,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) \
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) \
 		`cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/ModuleCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/ModuleCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/ModuleCodec/Makefile.in	2009-11-08 02:23:56.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/ModuleCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -17,7 +17,7 @@
 	chmod +x $@
 
 else
-	$(CXX) -fPIC -shared -o $@ $(OBJS) -Wl,--whole-archive dumb/lib/unix/libdumb.a \
+	$(CXX) $(CXXFLAGS) -shared -o $@ $(OBJS) -Wl,--whole-archive dumb/lib/unix/libdumb.a \
 		-Wl,--no-whole-archive `cat ../../DllLoader/exports/wrapper.def` \
 		../../DllLoader/exports/wrapper.o
 
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/NSFCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/NSFCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/NSFCodec/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/NSFCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -20,7 +20,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
 include ../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/SIDCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/SIDCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/SIDCodec/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/SIDCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -17,7 +17,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) ../../../linux/XHandle.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) ../../../linux/XHandle.o `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
 include ../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU/Makefile.in	2009-03-22 19:56:49.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -1,8 +1,8 @@
 ARCH=@ARCH@
 AS=nasm
 OBJS=SPC700.o APU.o DSP.o SNESAPU.o
-CFLAGS +=-D_LINUX -fPIC -I../../ -O2
-CXXFLAGS +=-D_LINUX -fPIC -I../../ -O2
+CFLAGS +=-D_LINUX -fPIC -I../..
+CXXFLAGS +=-D_LINUX -fPIC -I../..
 ifeq ($(findstring osx,$(ARCH)), osx)
 	CFLAGS+=-fno-common
 	CXXFLAGS+=-fno-common
@@ -25,7 +25,7 @@
 	../../../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -shared -o $@ $(OBJS) \
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) \
 		`cat ../../../../DllLoader/exports/wrapper.def` ../../../../DllLoader/exports/wrapper.o
 endif
 
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/timidity/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/timidity/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/timidity/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/timidity/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -1,6 +1,6 @@
 ARCH=@ARCH@
 
-CFLAGS+=-D_LINUX -fPIC -O2 -DHAVE_GETTIMEOFDAY -I. -Itimidity -Ilibarc -Iutils
+CFLAGS+=-D_LINUX -fPIC -DHAVE_GETTIMEOFDAY -I. -Itimidity -Ilibarc -Iutils
 ifeq ($(ARCH), powerpc-osx)
 	BUNDLE1_O=-lbundle1.o -L/usr/lib/gcc/powerpc-apple-darwin8/4.0.1/ -lgcc
 endif
@@ -57,7 +57,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CXX) -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o -Wl,--version-script=exports.def
+	$(CXX) $(CXXFLAGS) -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o -Wl,--version-script=exports.def
 endif
  
 include ../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/vgmstream/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/vgmstream/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/vgmstream/Makefile.in	2009-03-23 00:27:25.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/vgmstream/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -4,8 +4,8 @@
 	CFLAGS +=-D_LINUX -fPIC -DPIC -O3 -DXBMC
 	CXXFLAGS +=-D_LINUX -fPIC -DPIC -O3 -DXBMC
 else
-	CFLAGS +=-D_LINUX -fPIC -DPIC -O3 -fno-stack-protector -DXBMC
-	CXXFLAGS +=-D_LINUX -fPIC -DPIC -O3 -fno-stack-protector -DXBMC
+	CFLAGS +=-D_LINUX -fPIC -DXBMC
+	CXXFLAGS +=-D_LINUX -fPIC -DXBMC
 endif
 
 CODING_OBJS=src/coding/acm_decoder.o \
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/vorbisfile/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/vorbisfile/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/vorbisfile/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/vorbisfile/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -33,7 +33,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -shared -o $@ $(VORBIS_LINK_OPTS) $(OGG_LINK_OPTS) \
+	$(CC) $(CFLAGS) -shared -o $@ $(VORBIS_LINK_OPTS) $(OGG_LINK_OPTS) \
 		`cat ../../DllLoader/exports/wrapper.def` \
 		../../DllLoader/exports/wrapper.o
 endif
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/WavPackCodec/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/WavPackCodec/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/WavPackCodec/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/WavPackCodec/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -14,7 +14,7 @@
 	../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) `cat ../../DllLoader/exports/wrapper.def` ../../DllLoader/exports/wrapper.o
 endif
 
 include ../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/cores/paplayer/YMCodec/StSoundLibrary/Makefile.in xbmc-9.11-10.1/xbmc/cores/paplayer/YMCodec/StSoundLibrary/Makefile.in
--- xbmc-9.11-10.0/xbmc/cores/paplayer/YMCodec/StSoundLibrary/Makefile.in	2009-09-03 11:42:46.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/cores/paplayer/YMCodec/StSoundLibrary/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -17,7 +17,7 @@
 	../../../../../tools/Mach5/wrapper.rb $@;mv output.so $@
 	chmod +x $@
 else
-	$(CC) -fPIC -shared -o $@ $(OBJS) `cat ../../../DllLoader/exports/wrapper.def` ../../../DllLoader/exports/wrapper.o
+	$(CC) $(CFLAGS) -shared -o $@ $(OBJS) `cat ../../../DllLoader/exports/wrapper.def` ../../../DllLoader/exports/wrapper.o
 endif
 
 include ../../../../../Makefile.include
diff -Naur xbmc-9.11-10.0/xbmc/lib/libid3tag/Makefile.in xbmc-9.11-10.1/xbmc/lib/libid3tag/Makefile.in
--- xbmc-9.11-10.0/xbmc/lib/libid3tag/Makefile.in	2009-10-07 10:28:05.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/lib/libid3tag/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -16,7 +16,7 @@
 	chmod +x $@
 else
 $(SLIB): libid3tag/.libs/libid3tag.so
-	$(CXX) -shared -fPIC -O3 -o $@ libid3tag/.libs/*.o \
+	$(CXX) $(CXXFLAGS) -shared -fPIC -o $@ libid3tag/.libs/*.o \
 		`cat ../../cores/DllLoader/exports/wrapper.def` \
 		../../cores/DllLoader/exports/wrapper.o
 endif
diff -Naur xbmc-9.11-10.0/xbmc/visualizations/Goom/Makefile.in xbmc-9.11-10.1/xbmc/visualizations/Goom/Makefile.in
--- xbmc-9.11-10.0/xbmc/visualizations/Goom/Makefile.in	2009-03-23 01:04:05.000000000 +0100
+++ xbmc-9.11-10.1/xbmc/visualizations/Goom/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -2,7 +2,7 @@
 INCLUDES=-I. -I.. -I../../linux -I../../ -I../../../guilib -I../../../visualisations -Igoom2k4-0/src
 OBJS=Main.o
 SLIB=../../../visualisations/Goom.vis
-CXXFLAGS=-g -O3 -fPIC
+CXXFLAGS=-fPIC
 DIRS=goom2k4-0
 
 $(SLIB): $(OBJS) goom2k4-0/src/.libs/libgoom2.a
diff -Naur xbmc-9.11-10.0/xbmc/visualizations/XBMCProjectM/Makefile.in xbmc-9.11-10.1/xbmc/visualizations/XBMCProjectM/Makefile.in
--- xbmc-9.11-10.0/xbmc/visualizations/XBMCProjectM/Makefile.in	2009-07-09 02:00:27.000000000 +0200
+++ xbmc-9.11-10.1/xbmc/visualizations/XBMCProjectM/Makefile.in	2010-01-26 05:14:09.000000000 +0100
@@ -2,7 +2,7 @@
 INCLUDES=-I. -I.. -I../../linux -I../../ -I../../../guilib -I../../../visualisations
 OBJS=Main.o
 SLIB=../../../visualisations/ProjectM.vis
-CXXFLAGS=-O3 -g -fPIC
+CXXFLAGS=-fPIC
 DIRS=libprojectM
 
 ifeq ($(findstring osx,$(ARCH)), osx)
@@ -18,7 +18,7 @@
 	$(MAKE) -C libprojectM
 else
 $(SLIB): $(OBJS) libprojectM/libprojectM.so
-	$(CXX) ./libprojectM/CMakeFiles/projectM.dir/*.o -fPIC -shared $(CFLAGS) -o $(SLIB) $(OBJS) `cat ../../cores/DllLoader/exports/wrapper.def` ../../cores/DllLoader/exports/wrapper.o
+	$(CXX) $(CXXFLAGS) ./libprojectM/CMakeFiles/projectM.dir/*.o -fPIC -shared -o $(SLIB) $(OBJS) `cat ../../cores/DllLoader/exports/wrapper.def` ../../cores/DllLoader/exports/wrapper.o
 
 libprojectM/libprojectM.so: 
 	$(MAKE) -C libprojectM
