--- configure.in.orig	2010-07-19 01:52:39.417268110 -0400
+++ configure.in	2010-07-19 01:53:38.539142826 -0400
@@ -1828,26 +1828,6 @@
   fi
 ], [0])
 
-XB_CONFIG_MODULE([lib/libmodplug], [
-  if test "$host_vendor" = "apple" ; then
-    CFLAGS="$CFLAGS -fno-common" \
-    CXXFLAGS="$CXXFLAGS -fno-common" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic --disable-shared --enable-static
-  else
-    true
-  fi
-], [0])
-
-XB_CONFIG_MODULE([lib/libmicrohttpd], [
-  if test "$host_vendor" = "apple" ; then
-    CFLAGS="$CFLAGS -fno-common" \
-    CXXFLAGS="$CXXFLAGS -fno-common" \
-    ./configure MACOSX_DEPLOYMENT_TARGET=10.4 --with-pic --disable-shared --enable-static
-  else
-    true
-  fi
-], [0])
-
 XB_CONFIG_MODULE([lib/cpluff], [
   if test "$host_vendor" = "apple" ; then
     ./configure --disable-nls --with-libiconv-prefix=/opt/local --with-libintl-prefix=/opt/local --without-readline CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS"
