--- xbmc-9.11-b1/configure.in.config	2009-11-18 02:47:48.000000000 +0100
+++ xbmc-9.11-b1/configure.in	2009-11-25 20:25:05.973069389 +0100
@@ -560,7 +560,7 @@
 
 # External libdts
 if test "$use_external_libraries" = "yes" || test "$use_external_libdts" = "yes"; then
-  AC_CHECK_LIB([dts],          [main],, AC_MSG_ERROR($missing_library))
+  AC_CHECK_LIB([dca],          [main],, AC_MSG_ERROR($missing_library))
 
   AC_MSG_NOTICE($external_libdts_enabled)
   USE_EXTERNAL_LIBDTS=1
@@ -1054,6 +1054,8 @@
 if [[ $3 != "1" ]]; then
     if [[ -d $1 ]]; then
         pushd $1
+	[[ -f configure  ]] && chmod +x configure
+	[[ -f configure2 ]] && chmod +x configure2
         $2
         if [[ $? -ne 0 ]]; then
             popd
