--- Makefile.in.orig	2009-12-31 18:35:33.201505454 -0500
+++ Makefile.in	2009-12-31 18:36:29.747380027 -0500
@@ -416,7 +416,7 @@
 	$(MAKE) -C tools/TexturePacker/
 
 install-bin: xbmc.bin # developement convenience target
-	sudo install -D xbmc.bin $(prefix)/share/xbmc
+	sudo install -D xbmc.bin $(DESTDIR)$(libdir)/xbmc
 
 ifeq ($(findstring osx,$(ARCH)), osx)
 	# TODO: add osx install
@@ -424,55 +424,51 @@
 install: install-binaries install-arch install-datas install-web
 
 install-binaries: install-scripts
-	@echo "Copying XBMC binary to $(prefix)/share/xbmc/xbmc.bin"
-	@install -D xbmc.bin $(prefix)/share/xbmc/xbmc.bin
-	@install -D xbmc-xrandr $(prefix)/share/xbmc/xbmc-xrandr
+	@echo "Copying XBMC binary to $(DESTDIR)$(libdir)/xbmc/xbmc.bin"
+	@install -D xbmc.bin $(DESTDIR)$(libdir)/xbmc/xbmc.bin
+	@install -D xbmc-xrandr $(DESTDIR)$(libdir)/xbmc/xbmc-xrandr
 	@echo "You can run XBMC with the command 'xbmc'"
 endif
 
 install-arch:
-	@# Arch dependent files
-	@find system screensavers visualisations -regextype posix-extended -type f -not -iregex ".*svn.*|.*win32(dx)?\.vis|.*osx\.vis" -iregex ".*$(ARCH).*|.*\.vis|.*\.xbs" -exec install -D "{}" $(prefix)/share/xbmc/"{}" \; -printf " -- %-75.75f\r"	
+	@find system screensavers visualisations -regextype posix-extended -type f -not -iregex ".*svn.*|.*win32(dx)?\.vis|.*osx\.vis" -iregex ".*$(ARCH).*|.*\.vis|.*\.xbs" -exec install -D "{}" $(DESTDIR)$(libdir)/xbmc/"{}" \; -printf " -- %-75.75f\r"	
 
 install-scripts:
-	@install -D tools/Linux/xbmc.sh $(prefix)/bin/xbmc
-	@install -D tools/Linux/xbmc-standalone.sh $(prefix)/bin/xbmc-standalone
-	@install -D -m 0644 tools/Linux/FEH.py $(prefix)/share/xbmc/FEH.py
-	@install -D -m 0644 tools/Linux/xbmc-xsession.desktop $(prefix)/share/xsessions/XBMC.desktop
+	@install -D tools/Linux/xbmc.sh $(DESTDIR)$(bindir)/xbmc
+	@install -D tools/Linux/xbmc-standalone.sh $(DESTDIR)$(bindir)/xbmc-standalone
+	@install -D -m 0644 tools/Linux/FEH.py $(DESTDIR)$(libdir)/xbmc/FEH.py
+	@install -D -m 0644 tools/Linux/xbmc-xsession.desktop $(DESTDIR)$(prefix)/share/xsessions/XBMC.desktop
 
 install-datas: install-scripts
 	@echo "Copying support and legal files..."
 	@for FILE in `ls README.linux LICENSE.GPL *.txt`; do \
-		install -D -m 0644 "$$FILE" $(prefix)/share/xbmc/; done
+		install -D -m 0644 "$$FILE" $(DESTDIR)$(libdir)/xbmc/; done
 	@echo "Done!"
-	@echo "Copying system files to $(prefix)/share/xbmc"
-	@# Arch independent files
-	@find language media scripts sounds userdata visualisations system -regextype posix-extended -type f -not -iregex ".*svn.*|.*\.so|.*\.dll|.*\.pyd|.*python/.*\.zlib|.*\.vis" -exec install -D -m 0644 "{}" $(prefix)/share/xbmc/"{}" \; -printf " -- %-75.75f\r"
-	@# Skins
-	@find skin -regextype posix-extended -type f -not -iregex '.*/\.svn/.*|^skin/[^/]*/media/.*[^x][^b][^t]' -exec install -D -m 0644 '{}' $(prefix)/share/xbmc/'{}' \; -printf " -- %-75.75f\r"
-	@# Icons and links
-	@mkdir -p $(prefix)/share/applications $(prefix)/share/pixmaps
-	@cp -a tools/Linux/xbmc.png $(prefix)/share/pixmaps/
-	@cp -a tools/Linux/xbmc.desktop $(prefix)/share/applications/
+	@echo "Copying system files to $(DESTDIR)$(libdir)/xbmc"
+	@find language media scripts sounds userdata visualisations system -regextype posix-extended -type f -not -iregex ".*svn.*|.*\.so|.*\.dll|.*\.pyd|.*python/.*\.zlib|.*\.vis" -exec install -D -m 0644 "{}" $(DESTDIR)$(libdir)/xbmc/"{}" \; -printf " -- %-75.75f\r"
+	@find skin -regextype posix-extended -type f -not -iregex '.*/\.svn/.*|^skin/[^/]*/media/.*[^x][^b][^t]' -exec install -D -m 0644 '{}' $(DESTDIR)$(libdir)/xbmc/'{}' \; -printf " -- %-75.75f\r"
+	@mkdir -p $(DESTDIR)$(prefix)/share/applications $(DESTDIR)$(prefix)/share/pixmaps
+	@cp -a tools/Linux/xbmc.png $(DESTDIR)$(prefix)/share/pixmaps/
+	@cp -a tools/Linux/xbmc.desktop $(DESTDIR)$(prefix)/share/applications/
 
 install-livedatas:
 	@echo "Install Live CD datas in $(prefix) ..."
-	@mkdir -p $(prefix)/bin
-	@install -D tools/XBMCLive/diskmounter $(prefix)/bin
-	@install -D tools/XBMCLive/installXBMC $(prefix)/bin
-	@install -D tools/XBMCLive/runXBMC $(prefix)/bin
-	@install -D tools/XBMCLive/setAlsaVolumes $(prefix)/bin
+	@mkdir -p $(DESTDIR)$(bindir)
+	@install -D tools/XBMCLive/diskmounter $(DESTDIR)$(bindir)
+	@install -D tools/XBMCLive/installXBMC $(DESTDIR)$(bindir)
+	@install -D tools/XBMCLive/runXBMC $(DESTDIR)$(bindir)
+	@install -D tools/XBMCLive/setAlsaVolumes $(DESTDIR)$(bindir)
 
 install-web:
-	@mkdir -p $(prefix)/share/xbmc/web
-	@cp -r web/Project_Mayhem_III/* $(prefix)/share/xbmc/web
-	@find $(prefix)/share/xbmc/web -depth -name .svn -exec rm -rf {} \;
+	@mkdir -p $(DESTDIR)$(libdir)/xbmc/web
+	@cp -r web/Project_Mayhem_III/* $(DESTDIR)$(libdir)/xbmc/web
+	@find $(DESTDIR)$(libdir)/xbmc/web -depth -name .svn -exec rm -rf {} \;
 
 uninstall:
 	@echo "Removing XBMC..."
-	@rm -rf $(prefix)/share/xbmc $(prefix)/bin/xbmc
-	@rm -rf $(prefix)/bin/xbmc-standalone
-	@rm -rf $(prefix)/share/xsessions/XBMC.desktop
+	@rm -rf $(DESTDIR)$(libdir)/xbmc $(DESTDIR)$(bindir)/xbmc
+	@rm -rf $(DESTDIR)$(bindir)/xbmc-standalone
+	@rm -rf $(DESTDIR)$(prefix)/share/xsessions/XBMC.desktop
 	@echo "Done!"
 
 reallyclean:
