--- Makefile.in.orig	2010-09-14 18:08:19.973814197 -0400
+++ Makefile.in	2010-09-14 18:08:57.258416052 -0400
@@ -361,9 +361,6 @@
   endif
 endif
 	$(MAKE) -C xbmc/cores/paplayer/YMCodec/StSoundLibrary	
-ifeq ($(or $(findstring powerpc,$(ARCH)),$(findstring x86_64-linux,$(ARCH)),$(findstring arm, $(ARCH))),)
-	$(MAKE) -C xbmc/cores/paplayer/SPCCodec/SNES/SNESAPU
-endif
 imagelib: dllloader
 	$(MAKE) -C xbmc/lib/cximage-6.0
 
